archiva-modules/archiva-base/archiva-configuration/src/main/java/org/apache/archiva/configuration/util/PathMatcher.java

```diff
@@ -248,20 +249,30 @@ public static boolean matchPath(String pattern, String str) {
      */
     public static boolean matchPath(String pattern, String str,
                                     boolean isCaseSensitive) {
-        String[] patDirs = tokenizePathAsArray(pattern);
-        return matchPath(patDirs, tokenizePathAsArray(str), isCaseSensitive);
+        String[] patDirs = tokenizePathAsArray( pattern, false );
+        return matchPath(patDirs, tokenizePathAsArray( str, false ), isCaseSensitive);
     }
 
-
-    static String[] tokenizePathAsArray(String path) {
+    /**
+     * 
+     * @param path
+     * @param osspecific
+     * @return
+     */
+    static String[] tokenizePathAsArray(String path, boolean osSpecific) {
         Path root = null;
-        Path fsPath = Paths.get( path );
-
-        if ( fsPath.isAbsolute()) {
-            root = fsPath.getRoot( );
-            path = root.relativize( fsPath ).toString();
+        try 
+        {
+            Path fsPath = Paths.get( path );
+            if ( fsPath.isAbsolute() ) {
+                root = fsPath.getRoot();
+                path = root.relativize( fsPath ).toString();
+            }
+        } catch (InvalidPathException ipe ) 
+        {
+            // invalid path, windauze hate **/* 
         }
-        char sep = File.separatorChar;
+        char sep = osSpecific ? File.separatorChar : '/';
```

Comment:
In case of a pattern this was not good for windows **/*.sha for example was not tokeninzed

Reply:
Eric,
I added the PathMatcher class  to get rid of the ant dependency. But may be it is easier to use the glob path matcher of java.nio.
It used like this: `FileSystems.getDefault().getPathMatcher("glob:" + fileNamePattern)`

What do you think?